packages/presentation/lib/src/bloc/game_bloc.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:domain/domain.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:presentation/src/bloc/state.dart';
+import 'package:presentation/src/bloc/events.dart';
+
+const maxAttempts = 3;
+const initIsGuessTrue = false;
+const suggestedNumberInit = -1;
+
+class GameBloc extends Bloc<GameEvent, GameState> {
+  final GenerateGuessNumberUseCase generateGuessNumber;
+  final MakeAttemptUseCase makeAttempt;
+
+  int _suggestedNumber = suggestedNumberInit;
+
+  GameBloc({
+    required this.makeAttempt,
+    required this.generateGuessNumber,
+  }) : super(InitState()) {
+    on<MakeAttemptEvent>((event, emit) => emit(_makeAttempt()));
+    on<NewGameEvent>((event, emit) => emit(_newGame));
+    on<SetSuggestedNumberEvent>((event, emit) => _suggestedNumber = event.number);
+  }
+
+  GameState get _newGame => GameInProgressState(
+        guessNumber: generateGuessNumber(),
+        guessAttempts: maxAttempts,
+      );
+
+  GameState _makeAttempt() {
+    final currentState = (state as GameInProgressState);
+    final params = _createAttemptParams();
+    final attemptResult = makeAttempt(params);
+    switch (attemptResult.runtimeType) {
+      case LoseAttempt:
+        return LoseState();
+      case WinAttempt:
+        return WinState();
+      case FailAttempt:
+        return currentState.copyWith(
+            guessAttempts: (attemptResult as FailAttempt).attemptsRemain);
```

Comment:
just imho, it's better to use if with type casting here:

if (attemptResult is LoseAttempt) {
 ...
} else {
...
}